src/parser/mod.rs

```diff
@@ -7943,21 +7943,29 @@ impl<'a> Parser<'a> {
 
             let is_mysql = dialect_of!(self is MySqlDialect);
 
-            let (columns, partitioned, after_columns, source) =
-                if self.parse_keywords(&[Keyword::DEFAULT, Keyword::VALUES]) {
-                    (vec![], None, vec![], None)
-                } else {
-                    let columns = self.parse_parenthesized_column_list(Optional, is_mysql)?;
+            let (columns, partitioned, after_columns, source) = if self
+                .parse_keywords(&[Keyword::DEFAULT, Keyword::VALUES])
+            {
+                (vec![], None, vec![], None)
+            } else {
+                let columns = self.parse_parenthesized_column_list(Optional, is_mysql)?;
 
-                    let partitioned = self.parse_insert_partition()?;
+                let partitioned = self.parse_insert_partition()?;
 
-                    // Hive allows you to specify columns after partitions as well if you want.
-                    let after_columns = self.parse_parenthesized_column_list(Optional, false)?;
+                // make sure we are not about to consume a query
+                let mut after_columns = vec![];
+                match self.peek_nth_token(1).token {
+                    Token::Word(w) if w.keyword != Keyword::SELECT => {
```

Comment:
Checking for select seems incorrect here

What about a query like 
```sql
INSERT INTO tbla (cola) (VALUES (1))
```

The Hive manual https://cwiki.apache.org/confluence/display/hive/languagemanual+dml doesn't appear to say anything about this syntax

```
Standard syntax:
INSERT OVERWRITE TABLE tablename1 [PARTITION (partcol1=val1, partcol2=val2 ...) [IF NOT EXISTS]] select_statement1 FROM from_statement;
INSERT INTO TABLE tablename1 [PARTITION (partcol1=val1, partcol2=val2 ...)] select_statement1 FROM from_statement;
 
Hive extension (multiple inserts):
FROM from_statement
INSERT OVERWRITE TABLE tablename1 [PARTITION (partcol1=val1, partcol2=val2 ...) [IF NOT EXISTS]] select_statement1
[INSERT OVERWRITE TABLE tablename2 [PARTITION ... [IF NOT EXISTS]] select_statement2]
[INSERT INTO TABLE tablename2 [PARTITION ...] select_statement2] ...;
FROM from_statement
INSERT INTO TABLE tablename1 [PARTITION (partcol1=val1, partcol2=val2 ...)] select_statement1
[INSERT INTO TABLE tablename2 [PARTITION ...] select_statement2]
[INSERT OVERWRITE TABLE tablename2 [PARTITION ... [IF NOT EXISTS]] select_statement2] ...;
 
Hive extension (dynamic partition inserts):
INSERT OVERWRITE TABLE tablename PARTITION (partcol1[=val1], partcol2[=val2] ...) select_statement FROM from_statement;
INSERT INTO TABLE tablename PARTITION (partcol1[=val1], partcol2[=val2] ...) select_statement FROM from_statement;
```

I may be missing something

Reply:
It does look like `INSERT INTO tbla (cola) (VALUES (1))` is valid syntax for Athena / Presto as well.  Maybe it needs to be broadened?

Reply:
From reading those docs, it looks like the PARTITION keyword is required.  I moved the after_columns parsing into that statement.  I don't know why I didn't do that originally, but I think this should resolve the situation.  I added another test case

Reply:
I may have mis-communicated this.  I went ahead and confirmed that the failing test case is incorrect.

Test case:
`
    #[test]
    fn test_replace_into_select() {
        let sql = r#"REPLACE INTO t1 (a, b, c) (SELECT * FROM t2)"#;

        assert!(Parser::parse_sql(&GenericDialect {}, sql).is_err());
    }
`

I created a local mysql table and successfully ran these commands.

`
mysql> INSERT INTO tab (col) (SELECT col FROM tab);
Query OK, 1 row affected (0.02 sec)
Records: 1  Duplicates: 0  Warnings: 0

mysql> REPLACE INTO tab (col) (SELECT col FROM tab);
Query OK, 2 rows affected (0.02 sec)
Records: 2  Duplicates: 0  Warnings: 0
`

However, you can not wrap values in mysql, copied below.  From what I can tell, only Trino / Athena allow values statement to be wrapped.

`
mysql> INSERT INTO tab (col) VALUES (1);
Query OK, 1 row affected (0.02 sec)

mysql> INSERT INTO tab (col) (VALUES (1));
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '(1))' at line 1
`

Unfortunately, I don't have the time to check every Dialect that is supported, but I know that Athena / Trino will succeed and at least MySQL will fail.

In this case, do we only want to succeed a parenthesis values statement in common, and fail for every other dialect?  There seems to be another bug here